### PR TITLE
feat/PRO-3225/new-token-tiles

### PR DESCRIPTION
**Pull request by RanaBug**
## Description

- New Token Tiles design and FE implementation

## How Has This Been Tested?

- Unit tests and manual testing

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary by CodeRabbit

- **New Features**
  - Introduced a tile for displaying token market data, including token rows with detailed price, volume, liquidity, and transaction metrics.
  - Added components for rendering token logos (with blockchain overlays), customizable tile titles with optional decorators, and formatted timestamps with shortened time units.
  - Enabled navigation from token rows when links are provided.
  - Added visual indicators for price direction with colored icons and percentage changes.

- **Style**
  - Added a new green color to the theme for market data highlights.

- **Bug Fixes**
  - Improved image fallback handling for token and chain logos to prevent broken images.

- **Tests**
  - Added comprehensive tests for token market data row components and the overall market data tile, ensuring correct rendering and edge case handling.

### CODE REVIEW

File: src/apps/pillarx-app/components/TileTitle/TitleTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { useState } from 'react';
+
+// components
+import Body from '../Typography/Body';
+
+type TileTitleProps = {
+  title?: string;
+  leftDecorator?: string;
+  rightDecorator?: string;
+};
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pillarx-app/components/TileTitle/TitleTitle.tsx:10]: Not the same props

File: src/apps/pillarx-app/components/TileTitle/TitleTitle.tsx
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pillarx-app/components/TileTitle/TitleTitle.tsx:1]: Yes :) I thought this could be in the future a re-usable component for any Tile Title

File: src/apps/pillarx-app/components/TokenMarketDataRow/LeftColumnTokenMarketDataRow.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    .replace('second', 's')
+    .replace('minutes', 'min')
+    .replace('minute', 'min')
+    .replace('hours', 'h')
+    .replace('hour', 'h')
+    .replace('days', 'd')
+    .replace('day', 'd')
+    .replace('months', 'mo')
+    .replace('month', 'mo')
+    .replace(/(\d+)\s+(?=[a-zA-Z])/g, '$1');
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pillarx-app/components/TokenMarketDataRow/LeftColumnTokenMarketDataRow.tsx:42]: Sure! Done!

File: src/apps/pillarx-app/components/TokenMarketDataRow/LeftColumnTokenMarketDataRow.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+              src={CopyIcon}
+              alt="copy-token-address"
+              className="w-2.5 h-3"
+              onClick={(e) => e.stopPropagation()}
+            />
+          </CopyToClipboard>
+        )}
+      </div>
+      <div className="flex flex-wrap gap-2 mobile:gap-1.5">
+        {timestamp && (
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pillarx-app/components/TokenMarketDataRow/LeftColumnTokenMarketDataRow.tsx:69]: Good catch, it would crash if invalid, so I've added extra steps to validate the timestamp first and avoid a crash. If the timestamp is not valid, then it would simply not show on the tile.

File: src/apps/pillarx-app/components/TokenMarketDataRow/TokenMarketDataRow.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    <div
+      className={`flex w-full h-full items-center justify-between gap-2 py-3 border-b-[1px] border-[#25232D]
+    ${listNumber === 8 && 'desktop:border-b-0 tablet:border-b-0 mobile:border-b-0'} 
+    ${listNumber === 4 && 'desktop:border-b-0'}
+    ${data.link && 'cursor-pointer'}`}
+      onClick={() => (data.link ? navigate(`${data.link}`) : undefined)}
+    >
+      <div className="flex items-center flex-1 min-w-0">
+        <Body className="desktop:text-base tablet:text-base mobile:text-sm font-normal text-white/[0.5] mr-4 mobile:mr-2.5">
+          0{listNumber}
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pillarx-app/components/TokenMarketDataRow/TokenMarketDataRow.tsx:28]: Ok I have now made changes to acomodate this

File: src/apps/pillarx-app/components/TokensWithMarketDataTile/TokensWithMarketDataTile.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  const { data: tokensWithMarketData } = data || {};
+
+  const dataTokens = tokensWithMarketData as TokensMarketData | undefined;
+
+  // Limit the rows to the first 8 items
+  const limitedRows = dataTokens?.rows?.slice(0, 8) || [];
+
+  // Divide the rows into two separate columns
+  const leftColumn = limitedRows.slice(0, 4);
+  const rightColumn = limitedRows.slice(4, 8);
```
[the remark this answers is not in the document]
Comment (RanaBug) [reply src/apps/pillarx-app/components/TokensWithMarketDataTile/TokensWithMarketDataTile.tsx:27]: Ok I have now made changes to make this work
